quinn-proto/src/connection/mod.rs

```diff
@@ -201,9 +201,15 @@ where
             crypto,
             handshake_cid: loc_cid,
             rem_handshake_cid: rem_cid,
-            retire_cid_seq: 0,
-            cids_active_seq: HashSet::new(),
-            local_cid_len,
+            local_cid_state: CidState {
```

Comment:
I would recommend moving this into a `CidState::new()`, since there are quite a lot of initializations that are unrelated to the `Connection`